sql/2020/07_SEO/2019 Rewrites/raw_by_device.sql

```diff
@@ -0,0 +1,30 @@
+#standardSQL
+# Gather data from the body grouped by device
+# ### are lines removed so I can test on a different table
+
+SELECT
+###  client,
+  COUNT(0) AS total,
+
+  # hreflang
+  COUNTIF(raw_hreflang_count > 0) AS raw_has_hreflang,
+  ROUND(COUNTIF(raw_hreflang_count > 0) * 100 / COUNT(0), 2) AS pct_raw_has_hreflang,
+
+  # json-ld
+  COUNTIF(raw_jsonld_scripts_count > 0) AS raw_has_jsonld,
+  ROUND(COUNTIF(raw_jsonld_scripts_count > 0) * 100 / COUNT(0), 2) AS pct_raw_has_jsonld
+
+FROM (
+  SELECT
+###   client,
+    #REGEXP_EXTRACT(body, '(?i)<title>([^(</title>)]*)</title>') AS titletag,
+    #REGEXP_EXTRACT(body, '(?i)<h1>([^(</h1>)]*)</h1>') AS h1,
+    ARRAY_LENGTH(REGEXP_EXTRACT_ALL(body, '(?i)<link[^>]*hreflang=[\'"]?([^\'"\\s>]+)')) AS raw_hreflang_count,
+    ARRAY_LENGTH(REGEXP_EXTRACT_ALL(body, '(?i)<script[^>]*type=[\'"]?application\\/ld\\+json[\'"]?[^>]*>(.*?)<\\/script>')) AS raw_jsonld_scripts_count
+  FROM
+    `httparchive.almanac.response_bodies_desktop_1k`  # should be httparchive.almanac.summary_response_bodies and uncomment ###s real one costs a lot!
+###  WHERE
+###    firstHtml
+    )
+###GROUP BY
+###  client
```

Comment:
Why's all this commented out? Can we delete or uncomment?

Also does it not need a date?